scipy/stats/tests/test_hypotests.py

```diff
@@ -159,3 +160,79 @@ def test_callable_cdf(self):
         r1 = cramervonmises(x, distributions.beta.cdf, args)
         r2 = cramervonmises(x, "beta", args)
         assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
+
+
+class TestCvm_2samp(object):
+    def test_invalid_input(self):
+        x = np.arange(10).reshape((2, 5))
+        y = np.arange(5)
+        assert_raises(ValueError, cramervonmises_2samp, x, y)
+        assert_raises(ValueError, cramervonmises_2samp, y, x)
+        assert_raises(ValueError, cramervonmises_2samp, [], y)
+        assert_raises(ValueError, cramervonmises_2samp, y, [1])
+        assert_raises(ValueError, cramervonmises_2samp, y, y, 'wrong_method')
+
+    def test_list_input(self):
+        x = [2, 3, 4, 7, 6]
+        y = [0.2, 0.7, 12, 18]
+        r1 = cramervonmises_2samp(x, y)
+        r2 = cramervonmises_2samp(np.array(x), np.array(y))
+        assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
+
+    def test_example_conover(self):
+        # Example 2 in Section 6.2 of W.J. Conover: Practical Nonparametric
+        # Statistics, 1971.
+        x = [7.6, 8.4, 8.6, 8.7, 9.3, 9.9, 10.1, 10.6, 11.2]
+        y = [5.2, 5.7, 5.9, 6.5, 6.8, 8.2, 9.1, 9.8, 10.8, 11.3, 11.5, 12.3, 12.5, 13.4, 14.6]
+        r = cramervonmises_2samp(x, y)
+        assert_almost_equal(r.statistic, 0.262, decimal=3)
```

Comment:
[NumPy suggests](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html) `assert_allclose`.

Is this exact or asymptotic? Looks like this is an exact test because `y` has more than 10 elements? Consider making it explicit.

Reply:
I am not sure how 0.18 is computed in the example. I compare it against the asymptotic value (since y has more than 10 elements). The exact value should be pretty close given the sample size

Reply:
I wasn't able to find that reference, so I couldn't confirm from that. I also couldn't find a good 2-sample CVM implementation in R, so I whipped up my own independently, and the two agree exactly. 
```python3
def cvm2samp_nofrills(x, y):
    # no frills, independent implementation of cvm 2-sample test
    n, m = len(x), len(y)
    x, y = np.sort(x), np.sort(y)  # ...one orders all the observations
    z = np.hstack((x, y))
    ranks = stats.rankdata(z)      # ...determines the ranks
    ri, sj = ranks[:n], ranks[n:]
    i, j = np.arange(1, n+1), np.arange(1, m+1)
    U = n*np.sum((ri - i)**2) + m*np.sum((sj - j)**2)   # [2] Equation 10
    T = U/(n*m*(n+m)) - (4*m*n-1)/(6*(m+n))             # [2] Equation 9
    eT = 1/6 + 1/(6*(m+n))                              # [2] Equation 11
    varT = (1/45*(m+n+1)/(m+n)**2                       # [2] Equation 14
            *((4*m*n*(m+n))-3*(m**2+n**2)-2*m*n)
            /(4*m*n))
    nT = (T - eT)/(45*varT)**.5 + 1/6                   # [2] Equation 15
    p = 1 - _cdf_cvm_inf(nT)  # limiting distribution already reviewed
    return T, p
T, p = cvm2samp_nofrills(x, y)
res = cramervonmises_2samp(x, y, method='asymp')
assert_allclose(res.statistic, T)
assert_allclose(res.pvalue, p)
```